src/drivers/dw/dma.c

```diff
@@ -1179,13 +1193,30 @@ static void dw_dma_process_block(struct dma_chan_data *chan,
 		chan->status = COMP_STATE_PREPARE;
 	}
 
-	chan->lli_current->ctrl_hi &= ~DW_CTLH_DONE(1);
-	dcache_writeback_region(chan->lli_current, sizeof(*chan->lli_current));
-
-	chan->lli_current = (struct dw_lli2 *)chan->lli_current->llp;
+	if (ll_uncached->ctrl_hi & DW_CTLH_DONE(1)) {
+		ll_uncached->ctrl_hi &= ~DW_CTLH_DONE(1);
+		chan->lli_current = (struct dw_lli2 *)chan->lli_current->llp;
+	}
 }
 #endif
 
+static int dw_dma_copy(struct dma *dma, int channel, int bytes, uint32_t flags)
```

Comment:
Best to have the whole function in the LLI ifdef since it does not do anything in non LLI mode OR have 2 copies of this function, one for LLI mode and the other non LLI mode. The aim is to reduce the number of ifdefs